tests/post/steps/test_ingress.py

```diff
@@ -283,6 +324,36 @@ def server_does_not_respond(host, context):
     assert isinstance(context["exception"], requests.exceptions.ConnectionError)
 
 
+@then(
+    parsers.re(
+        r"an (?P<protocol>HTTPS?) request (?:on path '(?P<path>[^']+)' )?"
+        r"on port (?P<port>\d+) on a (?P<plane>.*) IP returns "
+        r"(?P<status_code>\d+) '(?P<reason>.+)'"
+    ),
+    converters=dict(status_code=int),
+)
+def server_request_returns(
+    host, context, protocol, port, plane, path, status_code, reason
+):
+    perform_request(
+        host=host, context=context, protocol=protocol, port=port, plane=plane, path=path
+    )
+    server_returns(host=host, context=context, status_code=status_code, reason=reason)
+
+
+@then(
+    parsers.re(
+        r"an (?P<protocol>HTTPS?) request (?:on path '(?P<path>[^']+)' )?"
+        r"on port (?P<port>\d+) on a (?P<plane>.*) IP should not return"
+    ),
+)
+def server_request_returns(host, context, protocol, port, plane, path):
```

Comment:
Same function name as above, I guess that's OK for pytest-bdd, but it may cause issues with static analysis (if we were to do that :sweat_smile:)

```suggestion
def server_request_does_not_return(host, context, protocol, port, plane, path):
```

Reply:
good catch just copy/paste